extensions/named-modes.md

```diff
@@ -0,0 +1,356 @@
+---
+title: IRCv3 Named Modes Extension
+layout: spec
+copyrights:
+  -
+    name: "Attila Molnar"
+    period: "2014"
+    email: "attilamolnar@hush.com"
+  -
+    name: "Sadie Powell"
+    period: "2017"
+    email: "sadie@witchery.services"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the unprefixed `named-modes` capability name. Instead, implementations SHOULD use the `draft/named-modes` capability name to be interoperable with other software implementing a compatible work-in-progress version. The final version of the specification will use an unprefixed name.
+
+## Introduction
+
+This specification describes a mechanism to advertise and manipulate additional modes (channel and user configuration settings) beyond those specified in the original IRC RFCs.
+
+## Motivation
+
+The IRC protocol originally had a `MODE` command to configure channels, using single character as configuration keys.
+Implementations extend the base protocol with their own modes, causing some of them to exhaust the namespace available. Additionally, independent implementations can accidentally implement the same feature with different characters, or use the same characters for different features.
+
+This specification aims to resolve both these issues, by introducing a virtually infinite namespace for new configuration keys, and by offering a way for implementations to define their own sub-namespaces via vendor prefixes to avoid clashes.
+
+## Capability
+
+The `draft/named-modes` capability allows clients to indicate support for receiving the new commands and numerics defined in this specification. Clients MUST negotiate this capability before using this functionality.
+
+## New numerics on connection
+
+These numerics MAY occur more than once. If the reply consists of multiple lines (due to IRC length limitations) all but the last numeric MUST have a parameter containing only an asterisk (`*`) preceding the mode list.
+
+### RPL_CHMODELIST
+
+    :<server name> XXX <nick> [*] {<type>:<modename>[=<letter>]}+
+
+where `<type>` is one of the following that tells the client about the nature
+of the mode:
+
+* 1 - mode is a list mode, i.e. requires a parameter when setting and unsetting (group 1 in 005 CHANMODES).
+* 2 - mode is a parameter mode, i.e. requires a parameter when setting and unsetting (group 2 in 005 CHANMODES).
+* 3 - mode is a parameter mode, i.e. requires a parameter when setting and no parameter when unsetting (group 3 in 005 CHANMODES).
```

Comment:
This reads strange. "a parameter mode" is being defined in two different ways but with the same nomenclature. Can we add something to differentiate them before the "i.e."?

Maybe just replace ", i.e." in all 5 of these with "that". e.g.

> * 2 - mode is a parameter mode that requires a parameter when setting and unsetting (group 2 in 005 CHANMODES).